task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -501,6 +503,22 @@ def run(ti: RuntimeTaskInstance, log: Logger):
         # TODO: Get a real context object
         ti.hostname = get_hostname()
         ti.task = ti.task.prepare_for_execution()
+        if ti.task.inlets or ti.task.outlets:
+            inlets = [
+                AssetProfile(name=x.name or None, uri=x.uri or None, asset_type=Asset.__name__)
+                for x in ti.task.inlets
+                if isinstance(x, Asset)
```

Comment:
Is this worth adding an `asprofile` or something method on Asset so we can do

```suggestion
                asset.asprofile() for asset in ti.task.inlets if isinstance(asset, Asset)
```

Reply:
Yeah, seems its worth it.
Added in the last batch of commits

Reply:
This is done!